File: src/frontend/src/app/types/family.ts
export interface FamilyMember {
  id: string
  familyId: string
  userId: string
  firstName: string
  lastName: string
  email: string
  phoneNumber: string | null
  isAdmin: boolean
  color: string
  joinedAtUtc: string
}

export interface Family {
  id: string
  name: string
  createdByUserId: string
  createdAtUtc: string
  members: FamilyMember[]
}

export interface CreateFamilyRequest {
  name: string
}

export interface AddFamilyMemberRequest {
  firstName: string
  lastName: string
  email: string
  phoneNumber?: string | null
  isAdmin?: boolean
  color?: string | null
}

export interface UpdateFamilyMemberRequest {
  firstName: string
  lastName: string
  email: string
  phoneNumber?: string | null
  isAdmin: boolean
}

export interface UpdateFamilyRequest {
  name: string
}

File: src/ios/FamilyHub/Models/Family.swift
import Foundation
import SwiftUI

struct FamilyMember: Codable, Identifiable, Equatable {
    let id: UUID
    let familyId: UUID
    let userId: UUID
    let firstName: String
    let lastName: String
    let email: String
    let color: String
    let joinedAtUtc: Date

    var fullName: String { "\(firstName) \(lastName)" }

    var initials: String {
        "\(firstName.prefix(1))\(lastName.prefix(1))".uppercased()
    }

    var swiftUIColor: Color {
        Color(hex: color) ?? .accentColor
    }
}

struct Family: Codable, Identifiable, Equatable {
    let id: UUID
    let name: String
    let createdByUserId: UUID
    let createdAtUtc: Date
    let members: [FamilyMember]

    var initials: String {
        String(name.prefix(2)).uppercased()
    }
}

struct CreateFamilyRequest: Encodable {
    let name: String
}

struct AddFamilyMemberRequest: Encodable {
    let email: String
}
